.github/workflows/Bonsai.yml

```diff
@@ -0,0 +1,460 @@
+      - name: Update version and submodule in ${{vars.DOCS_REPO}}
+        if: github.event_name == 'release' && github.event.release.prerelease == false
+        run: gh workflow run --repo "$DOCS_REPO" version-bump.yml --raw-field "project=$PROJECT" --raw-field "version=$VERSION" --raw-field "project-fork-url=$PROJECT_FORK_URL"
+        env:
+          GH_TOKEN: ${{secrets.DOCS_UPDATE_GH_TOKEN}}
+          DOCS_REPO: ${{vars.DOCS_REPO}}
+          PROJECT: bonsai
+          VERSION: ${{github.event.release.tag_name}}
+          PROJECT_FORK_URL: ${{github.server_url}}/${{github.repository}}.git
+
+  # =====================================================================================================================================================================
+  # Finish up release
+  # =====================================================================================================================================================================
+  finish-up-release:
+    name: Finish up release
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to bump tags and version numbers
+      contents: write
+      # Needed to close milestones
+      issues: write
+    # It's important that this only happens when everything else happened successfully
+    # Otherwise it's possible for a partial release to occur and re-running the job wouldn't be possible due to the latest tag having already moved
+    # (This is also why this happens as its own job, it lets us ensure both publish paths happened successfully.)
+    needs: [publish-github, publish-packages-nuget-org]
+    # Only do finishing tasks when we're explicitly releasing
+    if: github.event_name == 'release' && github.event.release.prerelease == false
+    steps:
+      # ----------------------------------------------------------------------- Checkout
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: refs/heads/main
+
+      # ----------------------------------------------------------------------- Update `latest` tag
+      - name: Update `latest` tag
+        run: git push --force origin ${{github.sha}}:refs/tags/latest
+
+      # ----------------------------------------------------------------------- Get the revision of the main branch
+      # We can't easily bump the version number unless the release happened at main
+      - name: Get the revision of the main branch
+        id: main-revision
+        run: python .github/workflows/gha.py set_output sha `git rev-parse refs/heads/main`
+
+      # ----------------------------------------------------------------------- Bump version number
+      - name: Warn if it doesn't match the release
+        if: steps.main-revision.outputs.sha != github.sha
+        run: python .github/workflows/gha.py print_warning "The main branch is at ${{steps.main-revision.outputs.sha}} but the release was made from ${{github.sha}}, the version number will not be automatically bumped."
```

Comment:
Note to self: This might actually be an overly aggressive check

Reply:
We talked about this during our meeting, but I looked at it more now and I remembered why this exists.

This part of the workflow pushes commits, those commits have to go to some branch, we choose `main` since that's sensible.

However releases are made from tags, not branches. If someone were to make a release from another branch (EG: someone backporting a critical fix to Bonsai 2.9 after Bonsai 3.0 has released) we wouldn't want to commit the version bump to `main` since the release didn't happen from there.

The fact that this also skips version bumping when someone pushes commits to `main` during the workflow run isn't really intentional. There's no really a straightforward way to avoid that edge case though, and I don't think it's really worth handling. (If the Bonsai repo is ever so busy that this becomes a concern this automated version bumping will likely need to have been changed anyway.)

I will reword some comments to clarify that this is what the intent is here.

Reply:
Also it's worth noting that bumping the version number isn't super critical. It only impacts the version used for dev builds and unstable builds. For actual releases (and pre-releases) the version specified on the GitHub interface is always what's used.

(This is also why it doesn't really matter if the automatic version isn't a version we actually intend to release. For example it'll be 2.8.5 after we release 2.8.4 even though we probably won't actually release a 2.8.5)